main/src/main/scala/sbt/EvaluateTask.scala

```diff
@@ -146,12 +146,27 @@ final case class PluginData(
     scalacOptions: Seq[String]
 ) {
   val classpath: Seq[Attributed[File]] = definitionClasspath ++ dependencyClasspath
+
+  // Not a constructor parameter to avoid a binary incompatibility
+  private[this] var _buildSources: BuildSources = BuildSources.Empty
+  private[sbt] def setBuildSourcesData(data: BuildSources): PluginData = {
+    _buildSources = data; this
+  }
+  private[sbt] def buildSourcesData: BuildSources = _buildSources
```

Comment:
@eed3si9n Is it really necessary to preserve the binary compatibility of `PluginData`? How bad would it be to break it in sbt 1.6?

I am asking this because I also want to add more parameters in it for #6566.

Reply:
The breakage is just for the constructor, copy, and apply factory methods, not for consumers of the data. I think it is highly unlikely to break existing plugins.

Reply:
>  Is it really necessary to preserve the binary compatibility of `PluginData`? How bad would it be to break it in sbt 1.6?

We maintain binary compatibility for the purpose of maintaining plugin compatibility, so I think `PluginData` would be ok to break, but ultimately we the only way to quantify this would be to release a milestone and test it using Community Build. If none of the existing builds break, then we should be ok.